architecture/2025-04-compute-platform-extensibility.md

```diff
@@ -0,0 +1,505 @@
+# ACI Integration
+
+* **Author**: Brooke Hamilton
+
+## Overview
+
+The Radius and ACI integration aims to enable deployments of applications and containers using the Radius control plane in Kubernetes to target ACI environments. The primary goals include setting up Radius environments targeting ACI, deploying to ACI, and managing ACI-specific configurations through punch-through configurations, including confidential containers.
+
+This document describes the initial integration with ACI, which is built upon an earlier demo capability. Our longer-term approach will provide an extensibility capability to Radius for deploying to other compute platforms, which will be described in a separate document.
+
+## Terms and definitions
+
+* ACI: Azure Container Instances, a service that allows users to run Docker containers on Azure without managing servers.
+
+## Objectives
+
+* To integrate Azure Container Instances (ACI) with Radius, enabling users to deploy and manage their applications and containers in ACI environments.
+* To provide a seamless user experience for deploying Radius applications to ACI, including environment setup, application definition, and deployment.
+* To ensure that the initial features are fully functional and ready for demonstration.
+
+> **Issue Reference:** <!-- (If appropriate) Reference an existing issue that describes the feature or bug. -->
+
+### Goals
+
+* Implement the capability for the Radius control plane hosted in Kubernetes to deploy to ACI.
+* Enable users to set up a Radius Environment targeted at ACI using environment definition files (e.g., env.bicep).
+* Allow users to deploy Radius applications to ACI using application definition files (e.g., app.bicep).
+* Support the deployment of application containers to ACI, ensuring that container definitions are platform-agnostic.
+* Allow users to apply ACI-specific configurations to containers using extension properties.
+* Adding and configuring Dapr sidecars to ACI containers.
+* A default implementation of Azure Gateways on ACI.
+* Defining and creating Secret Stores for use in ACI containers. (The implementation is TBD: KeyVault or Dapr.)
+
+### Non goals
+
+* Hosting the Radius control plane in ACI. The control plane will continue to be hosted in Kubernetes.
+* Defining and creating custom ingress for use in ACI containers.
+* Defining Extenders/UDT for use in ACI containers.
```

Comment:
Would this be difficult considering we do the other resource types?

Reply:
This has changed - we will use extenders for confidential compute configuration. I will remove it as a non-goal.